addons/mail/models/res_partner.py

```diff
@@ -64,6 +64,23 @@ def _compute_im_status(self):
         if odoobot in self:
             odoobot.im_status = 'bot'
 
+    @api.depends_context('company')
+    @api.depends('company_id')
+    def _compute_partner_share(self):
+        """Make sure company contacts are treated as internal partner"""
+        super()._compute_partner_share()
+        for partner in self:
+            if partner.partner_share and partner.email:
+                partner.partner_share = False if (
+                    (
+                            partner.company_id.alias_domain_id and
+                            partner.company_id.alias_domain_id.name in partner.email
+                    ) or (
+                            self.env.company.alias_domain_id and
+                            self.env.company.alias_domain_id.name in partner.email
+                    )
```

Comment:
Can you explain the reasoning?

Whatever you're trying to achieve, this isn't the way